assets/tests/state/searchPageState.test.ts

```diff
@@ -148,6 +149,35 @@ describe("reducer", () => {
     })
   })
 
+  describe("setSearchProperties", () => {
+    test("ensures limit is nonzero for all properties in the given list", () => {
+      const oldState: SearchPageState = searchPageStateFactory.build({
+        query: emptySearchQueryFactory.build({
+          text: "123",
+          properties: { run: 100, vehicle: 0 },
+        }),
+      })
+
+      expect(oldState.query.properties).toEqual({
+        location: defaultResultLimit,
+        operator: defaultResultLimit,
+        vehicle: 0,
+        run: 100,
+      })
```

Comment:
**question:** as I understand it, this assertion is basically testing the factory functions. Do you think it's necessary or is the second assertion the main one that matters for this test?

Reply:
True, the second assertion is the one that matters! I think alternatively I'll just add location & operator to the defined search query so it is easy to tell that those properties have limits specified and are also affected.